Presentation.Web/Controllers/API/V2/Internal/KLE/KLEInternalV2Controller.cs

```diff
@@ -0,0 +1,158 @@
+
+            var list = new List<dynamic>();
+            CreateCsvHeader(list);
+            CreateCsvChangeDescriptions(list, result.Value);
+
+            return CreateCsvFormattedHttpResponse(list);
+        }
+
+        [HttpPut]
+        [Route("update")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.BadRequest)]
+        public IHttpActionResult PutKLEChanges()
+        {
+            var result = _kleApplicationService.UpdateKLE();
+
+            return result.Ok ?
+                Ok(
+                    new KLEUpdateResponseDTO
+                    {
+                        Status = result.Value.ToChoice()
+                    }) :
+                FromOperationFailure(result.Error);
+        }
+
+        #region Helpers
+
+        private static void CreateCsvHeader(ICollection<dynamic> list)
+        {
+            var header = new ExpandoObject() as IDictionary<string, object>;
+            header.Add(KLETypeColumnName, Constants.KLE.Type.ColumnName);
+            header.Add(KLETaskKeyColumnName, Constants.KLE.TaskKey.ColumnName);
+            header.Add(KLEDescriptionColumnName, Constants.KLE.Description.ColumnName);
+            header.Add(KLEChangeColumnName, Constants.KLE.Change.ColumnName);
+            header.Add(KLEChangeDetailsColumnName, Constants.KLE.ChangeDetails.ColumnName);
+            list.Add(header);
+        }
+
+        private void CreateCsvChangeDescriptions(ICollection<object> list, IEnumerable<KLEChange> kleChanges)
+        {
+            var relevantChanges = kleChanges.Where(c => c.ChangeType != KLEChangeType.UuidPatched);
+            foreach (var elem in relevantChanges)
+            {
+                var obj = new ExpandoObject() as IDictionary<string, object>;
+                obj.Add(KLETypeColumnName, elem.Type);
+                obj.Add(KLETaskKeyColumnName, elem.TaskKey);
+                obj.Add(KLEDescriptionColumnName, elem.UpdatedDescription);
+                obj.Add(KLEChangeColumnName, ChangeTypeToString(elem.ChangeType));
+                obj.Add(KLEChangeDetailsColumnName, elem.ChangeDetails);
+                list.Add(obj);
+            };
+        }
+
+        private static IHttpActionResult CreateCsvFormattedHttpResponse(IEnumerable<dynamic> list)
+        {
+            var s = list.ToCsv();
+            var bytes = Encoding.UTF8.GetBytes(s);
+            var stream = new MemoryStream();
+            stream.Write(bytes, 0, bytes.Length);
+            stream.Seek(0, SeekOrigin.Begin);
+            var result = new HttpResponseMessage(HttpStatusCode.OK)
+            {
+                Content = new StreamContent(stream),
+            };
+            result.Content.Headers.ContentType = new MediaTypeHeaderValue("text/csv");
+            result.Content.Headers.ContentDisposition = new ContentDispositionHeaderValue("attachment")
```

Comment:
Two string candidates for constants, especially "attachment" which is used twice